### ISSUE TITLE
About bots placement

### ISSUE THREAD

**Issue by Saiska** [trickerer/Trinity-Bots#122]
I ignored it since because it wasn't a problem which one shot the bots, usually, but alas syndragosa came and resolved me to talk about bots placement.
It's not about tank or melee, it's about healer/caster
melee go in the boss back, so usually they are ok. Anyway cleaves and other melee boss abilities rarely targets bots, so they survive.
It's another matter with dragons breaths. Bots usually take it full charge.

Problem is:
- Caster/Healer usually stay where they are or near at the beginning of the fight
- Dist 0 don't help if you tank, and also confuse the hunters which will go melee
- If you tank, healer tends to place themselves behind you

As the boss melee ability don't touch the bots, usually, like the tail on dragons, being behind is generaly ok, and i don't see any boss where it would be a problem to be behind for a caster, or at least, more than behind, not in front in the 90° cone which inccur a fatal death by any aoe magic, which, contrary to melee, usually affect the bots.
edit. Just now in ruby sanctum found a melee which affect bot (on the general dragon kin which double himself, hes depop i don't have the name...) maybe just check position on dragon/dragonkin ennemy type ?

**Comment by trickerer** [comment]
https://github.com/trickerer/TrinityCore-3.3.5-with-NPCBots/commit/c491040ba1adec847d62b6cc97c1ff47ec10d94f

I didn't test is on bosses

**Comment by Saiska** [comment]
testing on a icc run later, will follow up.

**Comment by Saiska** [comment]
Works great. 
They do lose some time for useless encounter, but well. The worst i got was an aggro from a mage (mages generate a lot of aggro, they do a very good dps but don't use mirror image or invisibility to reduce their aggro, so sometimes they take it, even with a 5sec before dps rule) and the boss began to turn away from the tank, and the entire raid began to be confused and started moved non stop, like the cac did before, but now full raid. Not a major problem imho.
They do however, like the dist 0, don't break their cast to move, and i'm sure some will die from that, but again, not a major problem. It's much more a problem on the dist 0 for the heals who move only when their gcd is waiting (i think)

**Comment by Saiska** [comment]
Found an interesting counter example on Halion.
Contrary to syndra, who land almost always the same so you can preapre and be on the side before, when you pull halion the tank don't turn it and you face it. Unfortunately, the caster stay in front a random time depending on their cast (like the shamelem stay forever, but the druidheal move almost direct, so i think its related to cast time ?) and all stay is the random: is the bots will be moving before or after the first breath ? and more than once it was the latter for at least some of them.
I think its related with my issue with bot heal lollygagging; at some degree, bot should feel 'urgency' to move rather than cast. And this 'urgency' during the engage should be very high.

**Comment by Saiska** [comment]
I managed halion by starting the fight with dist 0, tank sign on the boss, and take some seconds to place the raid on the side.

I did find another problem through, the hunters: if there is little to no place behind the boss (seen that on lana'thel) they go behind and melee, destroying their dps. even with combat distance set manually. It may be a bother, but i think either the code should look at the enemy race, as ive seen the problem only on dragons/dragonkins, or eventually a list of npc id in the config file.

Finally i profit of this  post to thank you again for these bots, i did manage to finish all the wotlk content in hard mode with little or no changes (ok, i deleted the phases in halion so i can have the 2 halion in the same room; i removed the frostmourne tp on LK because the bots are too slow to follow, and i reduced drastically the damage of the for aoe on the same LK because i lost healers, some changes in ulduar and none in colyseum, and removed the health gain of gutt in naxx, i would say it's not so much comparing to the complexity it was to play in raid 25 humans at the time), it was a great challenge. Thank you.
